consensus/core/src/core.rs

```diff
@@ -669,40 +737,48 @@ impl Core {
             }
             assert!(commits_until_update > 0);
 
-            // TODO: limit commits by commits_until_update, which may be needed when leader schedule length
-            // is reduced.
-            let decided_leaders = self.committer.try_decide(self.last_decided_leader);
+            // Always try to process the synced commits first
+            let mut decided_leaders = self
```

Comment:
one thing to consider here is that you moved some of the logic of leader schedule into universal committer where try_decide does not work like that and we kept the commit logic as a black box that just spits out decided leaders. What do you think about continuing to keep that separation of logic?

Reply:
Yeah my initial iteration was without moving part of that logic in universal committer and keeping it in Core. We can certainly do that. It's just that the truncation of the leaders is a bit more complex and decided to do everything in the `try_decide_synced` method.

Reply:
One suggestion is to remove `try_decide_certified` from universal committer and keep it as a separate method in core i.e. `get_certified_commit_leaders`. We are actually bypassing the committer with your new logic so it makes sense to not use the component at all.